core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -238,11 +256,51 @@ func (l *logPollerWrapper) LatestEvents() ([]evmRelayTypes.OracleRequest, []evmR
 			})
 		}
 	}
+	resultsReq = l.FilterPreviouslyDetectedRequests(resultsReq)
+	resultsResp = l.FilterPreviouslyDetectedResponses(resultsResp)
 
-	l.lggr.Debugw("LatestEvents: done", "nRequestLogs", len(resultsReq), "nResponseLogs", len(resultsResp), "nextBlock", nextBlock, "latest", latest)
+	l.lggr.Debugw("LatestEvents: done", "nRequestLogs", len(resultsReq), "nResponseLogs", len(resultsResp), "startBlock", startBlock, "endBlock", latest)
 	return resultsReq, resultsResp, nil
 }
 
+func (l *logPollerWrapper) FilterPreviouslyDetectedRequests(requests []evmRelayTypes.OracleRequest) []evmRelayTypes.OracleRequest {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	filteredRequests := []evmRelayTypes.OracleRequest{}
+	for _, request := range requests {
+		if _, ok := l.detectedRequests[request.RequestId]; !ok {
+			filteredRequests = append(filteredRequests, request)
+			l.detectedRequests[request.RequestId] = time.Now()
+		}
+	}
+	// Expire old requests if they are older than logPollerCacheDurationSec
+	for requestId, timeDetected := range l.detectedRequests {
```

Comment:
Again very inefficient. Iterating over the whole set just to prune oldest ones. You could have an additional data structure that is sorted by detected time (even if it's not perfect) and make it much faster.